cms/djangoapps/contentstore/views/item.py

```diff
@@ -1046,6 +1054,15 @@ def create_xblock_info(xblock, data=None, metadata=None, include_ancestor_info=F
         else:
             xblock_info["staff_only_message"] = False
 
+    if is_concise:
```

Comment:
It might be easier to refactor this method into two or more methods, because the logic is hard to read when it is this long.